### PR TITLE
WIP: Added metrics for Inventory count and connection count

### PR DESCRIPTION
**Pull request by Mrudraia**
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by rewantsoni** [comment]
@Mrudraia We are merging the changes to the `observability` branch, not the `enabled-metrics` branch, can you open the PR in that branch?

File: controllers/metrics.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	DBaasStackInstallationtHistogram.With(prometheus.Labels{metricLabelVersion: version}).Observe(duration.Seconds())
+}
+
+// SetPlatformStatus exposes dbaas_platform_status metric for each platform
+func SetPlatformStatusMetric(platformName dbaasv1alpha1.PlatformsName, status dbaasv1alpha1.PlatformsInstlnStatus, version string) {
+	if len(platformName) > 0 {
+		switch status {
+
+		case dbaasv1alpha1.ResultFailed:
+			DBaasPlatformInstallationtGauge.With(prometheus.Labels{metricLabelName: string(platformName), metricLabelStatus: string(status), metricLabelVersion: version}).Set(float64(0))
```
Comment (jianrongzhang89) [inline controllers/metrics.go:100]: Why do you set the value to 0?  ResultFailed is one of the labels.

File: controllers/metrics.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		case dbaasv1alpha1.ResultSuccess:
+			DBaasPlatformInstallationtGauge.Delete(prometheus.Labels{metricLabelName: string(platformName), metricLabelStatus: string(dbaasv1alpha1.ResultInProgress), metricLabelVersion: version})
+			DBaasPlatformInstallationtGauge.Delete(prometheus.Labels{metricLabelName: string(platformName), metricLabelStatus: string(dbaasv1alpha1.ResultFailed), metricLabelVersion: version})
```
Comment (jianrongzhang89) [inline controllers/metrics.go:103]: Why do you delete the metric? What about setting the value of this metric to 0?

File: controllers/metrics.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+			DBaasPlatformInstallationtGauge.With(prometheus.Labels{metricLabelName: string(platformName), metricLabelStatus: string(status), metricLabelVersion: version}).Set(float64(1))
+		case dbaasv1alpha1.ResultInProgress:
+			DBaasPlatformInstallationtGauge.With(prometheus.Labels{metricLabelName: string(platformName), metricLabelStatus: string(status), metricLabelVersion: version}).Set(float64(2))
```
Comment (jianrongzhang89) [inline controllers/metrics.go:106]: This metric and the one for ResultSuccess have different labels.  Why do you set the value to 2?

File: Makefile [hunk 4]
```diff
@@ -3,7 +3,7 @@
 # To re-generate a bundle for another specific version without changing the standard setup, you can:
 # - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
-VERSION ?= 0.2.0
+VERSION ?= 0.2.41
```
Comment (jianrongzhang89) [inline Makefile:6]: Please clean up all local build-related changes and unwanted commented lines of code.

File: controllers/metrics.go (hunk 1 above)
Comment (redhatHameed) [reply controllers/metrics.go:100]: added into help 
Help: "status of an installation of components and provider operators. values (success=1, failed=0, in progress=2) ",

**Comment by redhatHameed** [comment]
@Mrudraia can you add this PR into observability branch Thanks

**Comment by redhatHameed** [comment]
@Mrudraia I am closing this PR as discussed we are going with approach as PR #182 , you can open new PR for request time duration

**Pull request closed without merging**